### ISSUE TITLE
[UX]  integrate file editing into the file field widget

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#2810]
## Describe your issue or idea
This is a follow-up to https://github.com/backdrop/backdrop-issues/issues/2796 where @olafgrabienski suggested

> I'd suggest to integrate file editing to the node form, adding an Edit button, and opening the respective file edit form in a modal.

![screen-backdrop-pr-1955-b-image-edit-or-remove](https://user-images.githubusercontent.com/14188307/29715855-58cf2a00-89a9-11e7-9ca1-6dc3b6aff35e.png)

**Comment by olafgrabienski** [comment]
Thanks @jenlampton for filing this issue. I guess we can need some feedback here (tagging as such).

**Comment by klonos** [comment]
Yes, either a modal or an expanding/collapsing section that reveals the file editing options would be great.
